### DISCUSSION TITLE
yq fails to process files  in /tmp

### DISCUSSION THREAD

**Discussion by mxchist** [mikefarah/yq#1299] [General]
Hello. Decide not to open issue, 'cause there is may be my mistake

**Describe the issue**
The `yq` throws _Error: open /tmp/who_uses_this_service.yaml: no such file or directory_ error when I try to process a file placed in `/tmp` directory. Moreover, yq can process files in any descendant location of `/home` directory, but if you will copy this file in /tmp then yq will throw the same error.

Version of yq: 4.26.1
Operating system: `Linux 5.15.0-43-generic #46~20.04.1-Ubuntu SMP`
Installed via: snap

**How to reproduce**
In this example I will use `/tmp/who_uses_this_service.yaml`.
First, I check that there is already no `who_uses_this_service.yaml` in /tmp.
Then, check the current time: `date --rfc-3339=ns`:
_2022-08-10 01:59:13.065172044+03:00_

So, lets create file __who_uses_this_service.yaml__ in any descendant directory of /home:
```yaml
datacenter:
  dataline:
    crm-coupon:
      api:
        - 10.0.10.10
```
If you will query this file with yq, for example, `yq '.datacenter.dataline.crm-coupon.api[1]' who_uses_this_service.yaml`, you get the result (10.0.10.10).

So, now copy it in /tmp: `cp who_uses_this_service.yaml /tmp/who_uses_this_service.yaml`. Then run the same query, `yq '.datacenter.dataline.crm-coupon.api[1]' /tmp/who_uses_this_service.yaml`. You will get an error:
_Error: open /tmp/who_uses_this_service.yaml: no such file or directory_

Check the creation time and access mode:
_-rw-rw-r-- 1 mxchist mxchist 9907 2022-08-10 02:02:49.765757989 +0300 /tmp/pods.yaml_
File created, and its  creation time greater then the last `date --rfc-3339=ns` result. Both user and group is mxchist, my own. I can easily open and editing it.
May be, there is lack of access rights anyway? Try run it from within the superuser:
`sudo yq '.datacenter.dataline.crm-coupon.api[1]' /tmp/who_uses_this_service.yaml`
Nevertheless, still _Error: open /tmp/who_uses_this_service.yaml: no such file or directory_ throwing
I tried different files from different sources, and every file with its own query, and all the time the same result: works in normal directory, fails in /tmp.

**Comment by andry81** [comment]
The same issue under Cygwin environment in Windows: `Error: open /tmp/tmp.XXXXXXXXXX/...: The system cannot find the path specified.`
I've thought it somehow connected to the Cygwin shell itself, because there is nothing wrong with the `/tmp` directory credentials. They are just the same as for all other directories.

**Reply by mikefarah** [reply]
I think it's highly likely that this is a snap confinement issue see the readme for details. Have you tried download the binary from github and seeing if that works?

**Reply by andry81** [reply]
> I think it's highly likely that this is a snap confinement issue see the readme for details. Have you tried download the binary from github and seeing if that works?

You are right. If try to put the executable into cygwin `/bin`, then it tries to use `/tmp` (through the `mktemp`). But if directly use the windows style path, then it does work as expected. The Cygwin just does not convert it for a windows executable.

**Reply by mxchist** [reply]
That is. When remove `yq` in snap and install it with `wget` instead, files in `/tmp` began processing.
After this, also, invoking the yq throws error `bash: /snap/bin/yq: No such file or directory`. It happens while the current shell window not been closed, or `source` command not been invoked. Is it the snap issue, or yq's?

**Reply by mikefarah** [reply]
Seems like snap and maybe a bash PATH env issue to me.
